### ISSUE TITLE
Some Questions About DaNN

### ISSUE THREAD

**Issue by sunshining-bit** [jindongwang/transferlearning#226]
When I tested the model with my own data, I found that no matter whether the training loss "loss" increases the MMD loss "loss_mmd", the migration result of the source domain to the target domain does not change. This shows that MMD is not helpful to the result of domain adaptation.
        What parameters can be debugged in the code to highlight the effect of MMD on domain adaptation?
        The accuracy rate of the data provided by me is relatively high. The accuracy rate of the source domain prediction can reach about 96%, and the target domain migration prediction result is about 94% regardless of whether the MMD regular term is added or not. Will the effect of MMD decay in the case of high accuracy?
        Thank you.

**Comment by jindongwang** [comment]
Please provide more details of the problem, e.g., screen shot and your dataset description.
Normally you need to tune the hyperparameters.

**Comment by sunshining-bit** [comment]
&nbsp; &nbsp; Thanks for replying.
&nbsp; &nbsp; From my point of view, the basis for judging whether to add domain adaptation is whether to add the loss item of'loss_mmd'.
&nbsp; &nbsp; My data type is a tabular data set of .csv type.(Two categories)
&nbsp; &nbsp;&nbsp;The second column classifies (label) whether it is stable or not; the data after the fourth column are all feature quantities that characterize whether it is stable or not.



When the total loss of my mini-batch is loss=loss_c, I think that no domain adaptation item is added at this time. At this time, the prediction accuracy of the corresponding source domain should be high, and the prediction accuracy of the target domain should be low.
The following figure shows the test results of the last iteration:


&nbsp; &nbsp; When my mini-batch training total loss loss=loss_c+LAMBDA * loss_mmd, the test results are as follows:


&nbsp; &nbsp; From the comparison of the above two pictures, after adding loss_mmd, the training loss of the source domain is slightly increased, and the test loss of the target domain is slightly reduced, which is in line with the trend of domain adaptation after mmd is added. However, the difference between the two losses is too small, resulting in no change in the accuracy of the settlement.
&nbsp; &nbsp; How to modify to greatly improve the accuracy of the test target domain judgment?
&nbsp; &nbsp; I think that only the hyperparameters related to mmd can be adjusted, which are the following two hyperparameters:

&nbsp; &nbsp; But after I make the modification, the accuracy of the source domain and target domain may decrease at the same time.
&nbsp; &nbsp; How should I make adjustments, or do I need to replace the original training model of the single hidden layer feedforward neural network?
&nbsp; &nbsp; Do I need to try DAN or DDC?
&nbsp; &nbsp; Thank you.
------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "jindongwang/transferlearning"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2021年5月10日(星期一) 上午10:55
***@***.***&gt;;
抄送:&nbsp;"○冰o激ゞ凌 ***@***.******@***.***&gt;;
主题:&nbsp;Re: [jindongwang/transferlearning] Some Questions About DaNN (#226)





 
Please provide more details of the problem, e.g., screen shot and your dataset description.
 Normally you need to tune the hyperparameters.

**Comment by jindongwang** [comment]
@sunshining-bit Can't read your comments... So many error codes

**Comment by jindongwang** [comment]
如果你追求绝对精度，可以直接上DDC，相当于就是换一个网络。调参的话，主要调的参数有：learning rate, lambda, batch size等。

**Comment by sunshining-bit** [comment]
好的，谢谢东哥。

**Comment by onlinehuazai** [comment]
请问DANN这个域对抗损失一直下降不了，怎么回事
